### PR TITLE
fix/change for postgresql db

### PR DESCRIPTION
**Pull request by GlennLz**

### CODE REVIEW

**Comment by GlennLz** [comment]
All tests are green, but some column names might have to be changed in the database on the VPN.
a bunch of names have been changed so you might want to check if your code that isn't included in the tests still works as intended...

add .env and .env.test in root, point .env.test to test database

DB_HOST=
DB_USER=
DB_PASSWORD=
DB_DATABASE=
DB_PORT=
JWTSECRET=

'npm run test' will truncate the tables, add dummy data en run the tests on the test database.

**Comment by PandaGamersHDNL** [comment]
add the envs needed in the readme file

**Comment by DrunkenToast** [comment]
Is this ready for review? This branch has a lot of changes. Try to keep branches shorter next time.

File: src/validations/customer-validation.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {Customer} from '../classes/customer';
+import * as customerServices from '../services/customer-service';
+
+const errorList = {
+    customer:`A customer with this account already exists!`,
+}
+
+const checkUser = async (id:number): Promise<boolean> => {
+    if(!!await customerServices.getCustomerByUserId(id))
```
Comment (DrunkenToast) [inline src/validations/customer-validation.ts:9]: !! makes me anxious. Also you could just return the check.
Do a proper check here.

File: src/validations/customer-validation.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            throw new Error(errorList.customer);
+        }
+    } 
+    catch (error) 
+    {   
+        console.log(error);
+    }
+    
+    return errorVar;
+}
```
Comment (DrunkenToast) [inline src/validations/customer-validation.ts:32]: Same comment as before. Applies to a lot of other places it seems.

File: src/validations/employee-validation.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    roleID: `Given role is not valid!`
+}
+
+const checkNationalNumber = async (nationalNumber: string): Promise<boolean> => {
+    if(!!await userServices.getUserByNationalNumber(nationalNumber))
+    {
+        return false;
+    }
+    return true;
+}
```
Comment (DrunkenToast) [inline src/validations/employee-validation.ts:15]: "I want to validate if a national security number is correct by checking if anyone else has this same national security number" How does that make any sense?
Use a library or regex for this.

If this is function, inside the validation folder, is to check if it already exists then rename the function to indicate as such.

This also isnt a promise, just a bool return type.

File: src/validations/invoice-validation.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    supplierExist: `Supplier does not exist!`,
+    paymentStatus: `Invoice status is not valid!`,
+    invoicePeriode: `Period of the invoice is not valid!`,
+    creationDate: `Creation date is not valid!`,
+    startDate: `Start date is not valid!`
+}
+
+const unpaidStatus = 1;
+
+const checkCustomerExists = async ( id: number): Promise<boolean> => {
```
Comment (DrunkenToast) [inline src/validations/invoice-validation.ts:17]: Like this. (So function names are inconsistent).
Just return the check, also not a promise. This applies to all other cases.

File: tsconfig.json [hunk 5]
```diff
@@ -1,70 +1,70 @@
 {
```
Comment (DrunkenToast) [inline tsconfig.json:1]: erm?

**Review by DrunkenToast — CHANGES_REQUESTED** [review]
I don't agree with many design choices that were made with the API but we obviously don't have the time to overhaul the API. I suggest to take a look at the comments.
Don't add any new features to this branch. We need to get this merged for the front-end.

**Pull request closed without merging**